### ISSUE TITLE
Unexpected ArgumentListWrapping reported

### ISSUE THREAD

**Issue by gfreivasc** [pinterest/ktlint#1333]
The code reads as follows:

```kotlin
// Wrapper.kt
data class Wrapper(val int: Int)

private const val TOP = 11
private const val SIZE = 10

fun pi() {
    Observable.just(mutableListOf(Wrapper(TOP)))
        .startWith(
            Observable.just(
                MutableList(SIZE) {
                    Wrapper(it)
                }
            )
        )
        .share()
}
```

## Expected Behavior
Ktlint shouldn't notify no problems related to either `ArgumentListWrapping` or `Indentation`.

## Observed Behavior
detekt wrapping Ktlint reports:

```
ArgumentListWrapping - [] at /<...>/app/src/main/java/<...>/Wrapper.kt:13:13
ArgumentListWrapping - [] at /<...>/app/src/main/java/<...>/Wrapper.kt:18:5
```

And reformats the code to:

```kotlin
data class Wrapper(val int: Int)

private const val TOP = 11
private const val SIZE = 10

fun pi() {
    Observable.just(mutableListOf(Wrapper(TOP)))
        .startWith(
        Observable.just(
            MutableList(SIZE) {
                        Wrapper(it)
                    }
        )
    )
        .share()
}
```

Which now reports the following:

```
ArgumentListWrapping - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:13:13
ArgumentListWrapping - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:19:3
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:13:1
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:14:5
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:15:9
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:16:5
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:17:1
Indentation - [] at /<...>/app/src/main/java/com/<...>/Wrapper.kt:18:5
```

## Steps to Reproduce
Running with detekt, with detekt-formatting plugin (which wraps ktlint) using the following rules:

```yml
# configs.yml
formatting:
    active: true
    autoCorrect: true
    ArgumentListWrapping:
        active: true
        indentSize: 4
        maxLineLength: 100
    Indentation:
        indentSize: 4
        continuationIndentSize: 4
```

```
$ detekt --build-upon-default-config \
  --plugins PATH_TO/detekt-formatting.jar \
  --auto-correct \
  --config configs.yml \
  --input Wrapper.kt
```

## Your Environment
* Version of ktlint used: 0.42.1
* Wrapped by: detekt 1.19.0

**Comment by gfreivasc** [comment]
Same failed report occurs when we try to run `ktlint` 0.42.1 with `--experimental` (which enables this ArgumentListWrapping rule):

```
$ ktlint -F --experimental Wrapper.kt
/<...>/app/src/main/java/com/<...>/Wrapper.kt:13:13: Unexpected indentation (expected 8, actual 12)
/<...>app/src/main/java/com/<...>/Wrapper.kt:18:9: Unexpected indentation (expected 4, actual 8)
```

Will verify if newer versions still report this.

**Comment by gfreivasc** [comment]
It appears it doesn't happen on newer 0.43.2, closing this issue.

**Comment by gfreivasc** [comment]
It appears it doesn't happen on newer 0.43.2, closing this issue.
